### ISSUE TITLE
Feature discussion: integrate scaler fit properties into model

### ISSUE THREAD

**Issue by prekratko** [keras-team/keras#12862]
Remark: I don't know where to put this topic, so it's here. If it should be put somewhere else, I ask admin to let me know how to do it or let him do it.

This topic bothers me, probably as much as many other developers.
When I'm preparing data for training, among other steps I use a scaler, for example, MinMaxScaler to normalize the data.
Then I perform training and save model at some point. And program ends.
After a while, when I want to calculate predictions with another program, I have no idea how to normalize new data since my model has no idea that data has to be normalized. It's not aware of any scaler at all.
Therefore, I have to save scaler together with trained model. Those are two steps.

My conclusion is that, if I can save scaler fitted properties used on training data together with its trained model, then doing normalization of input data on future prediction calculations would be an easy task. Compared to now clumsy task of managing model and scaler separately.

If there are cases where people don't normalize their data, then for them this is irrelevant. But I suppose that majority of people do normalize their data for better training results. And therefore, integration of scaler properties with (its) trained model seems as a very useful feature.

**Comment by Dref360** [comment]
We voted on this feature last week. :) 
I don't think we have an ETA on when it comes out.


https://github.com/keras-team/governance/blob/master/rfcs/20190502-preprocessing-layers.md

**Comment by prekratko** [comment]
I have checked the page.
Normalization layer, will it make using scalers unnecessary? (I wish)
I mean, will normalization layer implement fit and transform in training phase and just transform in production use?
Or something else?
